index.js

```diff
@@ -31,6 +32,15 @@ function waitConnection (socket, write) {
   }
 }
 
+function isExternalUrl (url) {
+  try {
+    const u = new URL(url)
```

Comment:
unfortunately this is going to be very costly perf-wise. new URL is slow. I would recommend to use a simple regexp matching http and https prefixes.

Reply:
Got it, fixed as well.